### ISSUE TITLE
Admins can't login with version with v3.6.0 (API v2.12.1)

### ISSUE THREAD

**Issue by zbrag** [passbolt/passbolt_browser_extension#151]
## Admins can't login, users can
* Passbolt Version: 2.12.1
* Platform and Target:
-- Browser: Firefox 100.0.2
-- Browser: Chrome 101.0.4951.64

### What you did
The plugin was automatically updated to version 3.6.0

### What happened
After the update to 3.6.0 both of our admins couldn't login anymore. Not only the popup-extension wasn't working anymore, but the website version as well. Both browsers have this problem with the same plugin version. The browsers are used by two different individuals on two separate Mac OS X machines.

The error I got from the popup version was 'Could not validate entity externalGpgKey.'

### What you expected to happen
We expected every one to be able to use passbolt ;)

**Comment by gileri** [comment]
I think it's related to #150.

**Comment by AnatomicJC** [comment]
Hi,

Yes, it should be related :)

@zbrag: Can you check your passbolt gpg keys ?

```
# connect to your container
docker exec -it your_passbolt_container_name bash
# Check your server key
gpg --show-keys /etc/passbolt/gpg/serverkey.asc
```

If you are seeing more than one different PGP fingerprints like this:

```
pub rsa2048 2021-01-20 [SC]
177C1516F9C1957ABC157CA592D946CDEF1F7583
uid [ unknown] Passbolt default user <passbolt@yourdomain.com>
sub rsa2048 2021-01-20 [E]

pub rsa2048 2021-01-20 [SC]
DA5D26D04A2D7558EEF60E69C8DA4B7205E6E47F
uid [ unknown] Passbolt default user <passbolt@yourdomain.com>
sub rsa2048 2021-01-20 [E]

pub rsa2048 2021-01-20 [SC]
545F22A36F0380984D828F6BE5E5090C6FD6738A
uid [ unknown] Passbolt default user <passbolt@yourdomain.com>
sub rsa2048 2021-01-20 [E]

pub rsa2048 2021-01-20 [SC]
F9D5F72A0D7DB5118FE7146E9A1F3AC2510004EB
uid [ unknown] Passbolt default user <passbolt@yourdomain.com>
sub rsa2048 2021-01-20 [E]
```

You must rotate your key as described in this documentation: https://help.passbolt.com/faq/hosting/how-to-rotate-server-gpg-keys

Can you tell us if it fixes your issue ?

Cheers,

**Comment by zbrag** [comment]
My gpg doesn't seem to be having the --show-keys parameter, can you please tell me which command it is when using gpg (GnuPG) 2.0.22 / libgcrypt 1.5.3 ?

**Comment by AnatomicJC** [comment]
Hi @zbrag ,

Instead of --show-keys, can you try with `gpg --with-fingerprint /etc/passbolt/gpg/serverkey.asc` or `gpg --with-fingerprint /var/www/passbolt/config/gpg/serverkey.asc`.

On which platform are you running your passbolt server, docker, centos, debian, other ?

Cheers,

**Comment by zbrag** [comment]
@AnatomicJC That works! But unfortunatly my keys contain only one value (I'm not using docker, so I've looked at  the files indicated by passbolt->gpg-serverKey->private as well as the public.

Both of them only contain one key.
The server it's running on uses CentOS 4.1.

**Comment by cedricalfonsi** [comment]
As a workaround I would recommend your admins to perform a recover of their account with the backup of their private key they saved previously.

If it's not working or they want to help understand the issue, we would need an output of the browser extension debug console with the trace of the error. To do so:

On Chrome:
New tab > go to chrome://extensions > look for the passbolt card > details > inspect views section > click index.html > unfold and copy errors under console tab

On Firefox:
New tab > go to about:debugging#/runtime/this-firefox > look for the passbolt card > Click inspect > unfold and copy errors under console tab

**Comment by zbrag** [comment]
I'll try to recover the accounts later, these are the three errors I'm getting when I go passbolt (firefox):
`09:08:41.946
authPagemod::attach legacy account cannot be retrieved, please contact your administrator. [index.min.js:78661:17]
(moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js)
    onAttach moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:78661
    739]</__initConnectListener/this._listeners["chrome.runtime.onConnect" moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:79474
09:08:41.951 EntityValidationError: Could not validate entity externalGpgKey.
    EntityValidationError moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:53743
    validateObject moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:53408
    validate moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:53373
    ExternalGpgKeyEntity moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:61009
    findPublic moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:74428
    get moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:80273
    onAttach moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:78654
    739]</__initConnectListener/this._listeners["chrome.runtime.onConnect" moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:79474
[index.min.js:78662:17](moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js)
    onAttach moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:78662
    739]</__initConnectListener/this._listeners["chrome.runtime.onConnect" moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:79474
09:08:42.572
EntityValidationError: Could not validate entity externalGpgKey.
    EntityValidationError moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:53743
    validateObject moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:53408
    validate moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:53373
    ExternalGpgKeyEntity moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:61009
    findPublic moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:74428
    exec moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:43304
    _exec moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:43290
    listen moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:49615
    on moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:79688
[index.min.js:43293:15](moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js)
    _exec moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:43293
    listen moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:49615
    on moz-extension://85687f04-f094-0744-b1d8-9becc1816a78/index.min.js:79688

`
<img width="1317" alt="Screenshot 2022-06-09 at 09 11 32" src="https://user-images.githubusercontent.com/1644844/172787017-0fbcc681-2499-4581-ae14-bdd9972bea83.png">

If the chrome output is better/different, I can also provide it?

**Comment by cedricalfonsi** [comment]
Thank you for the additional information, it helped to narrow the issue.

We still miss few key elements to come up with a fix and strengthen the code.

Do you remember the version of the extension (or a date) you used when you configured or recovered your account on your browser? It will help us to identify the code that parsed your server public key, and stored the associated information in the local storage.

Some information regarding the server key in use prior to the issue would help too. I created a fiddle to extract the expires value: https://jsfiddle.net/xnt4jhfe/2/

Just copy/paste your server public key in the `armoredKey` variable value, run & look into the console.

**Comment by zbrag** [comment]
No, it has been at least 2 years since we've starting using Passbolt, I can find the generator version in the private key ( OpenPGP.js v2.6.2 ), does that help?

The fiddle seems to return 'Infinity' ?

**Comment by cedricalfonsi** [comment]
Thank you, it helps narrowing the issue.
Long story short, more validation was added on data retrieved from local storage. In some rare cases the expiry date of the openpgp keys stored in the local storage don't validate against the new constraints.
Recovering an account flushes the local storage and can be used as a workaround. A fix is on its way to take care of the issue and later problems relative to inconsistent expiry date.

**Comment by cedricalfonsi** [comment]
This should be fixed with v3.6.2.
Do not hesitate to reopen if the problem persists.
